arwen/host/execution_test.go

```diff
@@ -262,6 +264,91 @@ func TestExecution_ManyDeployments(t *testing.T) {
 	}
 }
 
+func TestExecution_MultipleArwens_OverlapingContractInstanceData(t *testing.T) {
+	code := GetTestSCCode("counter", "../../")
+
+	input := DefaultTestContractCallInput()
+	input.GasProvided = 1000000
+	input.Function = get
+
+	host1, instanceRecorder1 := defaultTestArwenForCallWithInstanceRecorderMock(t, code, nil)
+	runtimeContextMock := contextmock.NewRuntimeContextWrapper(&host1.runtimeContext)
+	runtimeContextMock.CleanWasmerInstanceFunc = func() {}
+	host1.runtimeContext = runtimeContextMock
+
+	vmOutput, err := host1.RunSmartContractCall(input)
+	require.Nil(t, err)
+	require.NotNil(t, vmOutput)
+
+	var host1InstancesData = make(map[interface{}]bool)
+	for _, instance := range instanceRecorder1.GetContractInstances(code) {
+		host1InstancesData[instance.GetData()] = true
+	}
+
+	host2, instanceRecorder2 := defaultTestArwenForCallWithInstanceRecorderMock(t, code, nil)
+	runtimeContextMock = contextmock.NewRuntimeContextWrapper(&host2.runtimeContext)
+	runtimeContextMock.CleanWasmerInstanceFunc = func() {}
+	host2.runtimeContext = runtimeContextMock
+
+	for i := 0; i < maxUint8+1; i++ {
+		vmOutput, err = host2.RunSmartContractCall(input)
```

Comment:
After each `host.RunSmartContractCall(input)`, you should be able to extract the `instance` itself from the usual `InstanceBuilderMock` like this:
```
host, _, ibm := defaultTestArwenForCallWithInstanceMocks(t)
vmOutput := host.RunSmartContractCall(input)
code := string(input.RecipientAddr)               // by convention, the code of an instance mock is its address
instance := ibm.InstanceMap[string(code)]      
```
This will allow a few simplifications:
* The `InstanceBuilderRecorderMock` might not be needed anymore
* The next `for` loop could be merged into this one.